### ISSUE TITLE
Scatter-gather with custom-aggregation-strategy

### ISSUE THREAD

**Issue by sleone** [mulesoft/munit#177]
Hi,
I recieve the following error if I try to use a custom-aggegation-strategy in a scatther-gather processor:

`nested exception is org.springframework.beans.factory.BeanDefinitionStoreException: Unexpected exception parsing XML document from class path resource [users.xml]; nested exception is org.mule.config.spring.parsers.processors.CheckExclusiveAttribute$CheckExclusiveAttributeException: The attribute 'class' cannot appear with the attribute '__MUNIT_LINE_NUMBER' in element custom-aggregation-strategy{__MUNIT_LINE_NUMBER=591, class=com.qvc.api.mule.aggregation.CustomAggregationStrategy}. (org.mule.api.lifecycle.InitialisationException)`

Can you help me please?
Thank you,
Sergio

**Comment by damiansima** [comment]
Hi there @sleone , 

I've just create an example of that and it seems to be working just fine, please check bellow:
**Production Code**
```xml
<?xml version="1.0" encoding="UTF-8"?>

<mule xmlns="http://www.mulesoft.org/schema/mule/core" xmlns:doc="http://www.mulesoft.org/schema/mule/documentation"
	xmlns:spring="http://www.springframework.org/schema/beans" version="EE-3.6.1"
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans-current.xsd
http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd">
    <flow name="logicFlow">
        <scatter-gather doc:name="Scatter-Gather">
            <custom-aggregation-strategy class="org.mule.munit.CustoAggregator"/>
            <set-payload value="#[1]" doc:name="Set Payload"/>
            <set-payload value="#[2]" doc:name="Set Payload"/>
            <set-payload value="#[3]" doc:name="Set Payload"/>
        </scatter-gather>
    </flow>
</mule>
```
**Custom Aggregator**
```java
package org.mule.munit;

import org.mule.DefaultMuleEvent;
import org.mule.api.MuleEvent;
import org.mule.api.MuleException;
import org.mule.api.routing.AggregationContext;
import org.mule.routing.AggregationStrategy;

public class CustoAggregator implements AggregationStrategy {

    @Override
    public MuleEvent aggregate(AggregationContext context) throws MuleException {
	MuleEvent e = DefaultMuleEvent.copy(context.getEvents().get(0));
	e.getMessage().setPayload("lalero");
	return e;
    }

}
```

**Test Code**

```xml
<?xml version="1.0" encoding="UTF-8"?>

<mule xmlns="http://www.mulesoft.org/schema/mule/core" xmlns:mock="http://www.mulesoft.org/schema/mule/mock" xmlns:munit="http://www.mulesoft.org/schema/mule/munit" xmlns:doc="http://www.mulesoft.org/schema/mule/documentation" xmlns:spring="http://www.springframework.org/schema/beans" xmlns:core="http://www.mulesoft.org/schema/mule/core" version="EE-3.6.1" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.mulesoft.org/schema/mule/mock http://www.mulesoft.org/schema/mule/mock/current/mule-mock.xsd
http://www.mulesoft.org/schema/mule/munit http://www.mulesoft.org/schema/mule/munit/current/mule-munit.xsd
http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans-current.xsd
http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd">
    <munit:config name="munit" doc:name="Munit configuration"/>
    <mock:config name="mock_logic" doc:name="Mock configuration"/>
    <spring:beans>
        <spring:import resource="classpath:logic.xml"/>
    </spring:beans>
    <munit:test name="logic-test-logicFlowTest" description="Test">
        <flow-ref name="logicFlow" doc:name="Flow-ref to logicFlow"/>
        <munit:assert-on-equals message="OOPS!" expected-ref="#['lalero']" value-ref="#[payload]" doc:name="Assert Equals"/>
    </munit:test>
</mule>

```
Then again, I guess you're trying to do something different. Could you please share your code ?
Cheers!

**Comment by sleone** [comment]
Hi @damiansima,
I tried to run your code and the result is the same error.

I'm using Mule 3.5.2 with Munit 3.5-M2.

Thanks for your help

**Comment by damiansima** [comment]
Hi @sleone, 

That changes things :P

So there was but in Mule 3.5.2, the one you reported. It was reported a few months ago and they fix it. 

But they fix it in Mule 3.5.3 (which will be release) and in 3.6.x. 
That's why it was working correctly for me I was using Mule 3.6.1 where the bug is also fixed as I said.

So the options you have hear are to move to a newer version of Mule. Either wait for 3.5.3 or just go ahead and user 3.6.1. 

While we are in the subject of versions, I will also suggest that you either use MUnit version 3.5-M3-SNAPSHOT or 3.6.0-BETA.
These two are more stable and up to date versions of MUnit. If you're working with Anypoint Studio just add this updatesite: http://studio.mulesoft.org/r4/munit

Cheers, Damian.
